slack-lint/src/test/java/slack/lint/resources/WrongResourceImportAliasDetectorTest.kt

```diff
@@ -0,0 +1,270 @@
+        @@ -4 +4
+        - import slack.l10n.R as L10R
+        + import slack.l10n.R as L10nR
+        @@ -9 +9
+        -         val appName = getString(L10R.string.app_name)
+        +         val appName = getString(L10nR.string.app_name)
+        """.trimIndent()
+      )
+  }
+
+  @Test
+  fun `test failure multiple references`() {
+    lint()
+      .files(
+        kotlin(
+          """
+          package slack.pkg.subpackage
+
+          import slack.l10n.R as L10R
+
+           class MyClass {
+
+             init {
+                  val appName = getString(L10R.string.app_name) + "-" + getString(L10R.string.suffix)
+                  getColor(R.color.transparent).let { println(it) }
+              }
+
+           }
+          """
+        )
+          .indented()
+      )
+      .run()
+      .expect(
+        """
+        src/slack/pkg/subpackage/MyClass.kt:3: Error: Please use L10nR as an import alias here [WrongResourceImportAlias]
+        import slack.l10n.R as L10R
+        ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        1 errors, 0 warnings
+        """.trimIndent()
+      )
+      .expectFixDiffs(
+        """
+        Autofix for src/slack/pkg/subpackage/MyClass.kt line 3: Replace import alias:
+        @@ -3 +3
+        - import slack.l10n.R as L10R
+        + import slack.l10n.R as L10nR
+        @@ -8 +8
+        -         val appName = getString(L10R.string.app_name) + "-" + getString(L10R.string.suffix)
+        +         val appName = getString(L10nR.string.app_name) + "-" + getString(L10nR.string.suffix)
+        """.trimIndent()
+      )
+  }
+
+  @Test
+  fun `test failure multiple wrong imports`() {
+    lint()
+      .files(
+        kotlin(
+          """
+          package slack.pkg.subpackage
+
+          import slack.uikit.resources.R as SKR
+          import slack.l10n.R as L10R
+
+           class MyClass {
+
+             init {
+                  getColor(SKR.color.transparent).let { println(it) }
+                  val appName = getString(L10R.string.app_name) + "-" + getString(L10R.string.suffix)
+              }
+
+           }
+          """
+        )
+          .indented()
+      )
+      .run()
+      .expect(
+        """
+        src/slack/pkg/subpackage/MyClass.kt:3: Error: Please use SlackKitR as an import alias here [WrongResourceImportAlias]
+        import slack.uikit.resources.R as SKR
+        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+        1 errors, 0 warnings
+        """.trimIndent()
+      )
+      .expectFixDiffs(
+        """
+        Autofix for src/slack/pkg/subpackage/MyClass.kt line 3: Replace import alias:
+        @@ -3 +3
+        - import slack.uikit.resources.R as SKR
+        + import slack.uikit.resources.R as SlackKitR
+        @@ -9 +9
+        -         getColor(SKR.color.transparent).let { println(it) }
+        +         getColor(SlackKitR.color.transparent).let { println(it) }
+        """.trimIndent()
```

Comment:
Shouldn't we expect a failure for `L10R` -> `L10nR` here as well?

Reply:
Nope. [WrongResourceImportAliasDetector](https://github.com/slackhq/slack-lints/pull/27/files#diff-430c4e9b542e289726fc7750126bb3f1ae2a266f7a01efbdc0b9db084d754d41R81) only reports the first wrong import alias to be able to offer a fix for it.
We could support multiple fixes by adding a map of `wrongAlias` to `fixes`, but that would complicate the detector even more.